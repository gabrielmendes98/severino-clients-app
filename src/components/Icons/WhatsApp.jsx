import * as React from 'react';
import PropTypes from 'prop-types';
import Svg, { Path } from 'react-native-svg';

const WhatsAppIcon = ({ size, ...props }) => (
  <Svg
    fill="none"
    height={size}
    viewBox="0 0 18 18"
    width={size}
    xmlns="http://www.w3.org/2000/svg"
    {...props}
  >
    <Path d="M14.25 3.945H3.737v9.538h10.515V3.945Z" fill="#fff" />
    <Path
      clipRule="evenodd"
      d="M14.894 2.497A8.71 8.71 0 0 0 8.754 0C3.968 0 .073 3.82.072 8.515c0 1.501.399 2.966 1.159 4.258L0 17.185l4.602-1.184a8.804 8.804 0 0 0 4.148 1.037h.003c4.784 0 8.679-3.82 8.68-8.516a8.383 8.383 0 0 0-2.539-6.025ZM5.228 8.825c-.108-.142-.885-1.154-.885-2.201 0-1.047.56-1.562.759-1.775a.804.804 0 0 1 .578-.266c.145 0 .29.001.416.007.133.007.312-.05.488.365.18.427.614 1.474.669 1.58.054.107.09.231.018.373s-.109.23-.217.355c-.108.124-.228.278-.325.373-.109.106-.222.221-.096.434.127.213.562.91 1.207 1.474.828.725 1.527.95 1.744 1.056.217.107.343.09.47-.053.127-.142.542-.621.687-.834.145-.213.29-.178.488-.107.199.071 1.265.586 1.482.692.217.107.361.16.416.25.054.088.054.514-.127 1.01-.18.498-1.047.952-1.464 1.012a3.025 3.025 0 0 1-1.366-.084 12.597 12.597 0 0 1-1.237-.448c-2.175-.922-3.596-3.07-3.705-3.213Z"
      fill="#4CAF50"
      fillRule="evenodd"
    />
  </Svg>
);

WhatsAppIcon.defaultProps = {
  size: 18,
};

WhatsAppIcon.propTypes = {
  size: PropTypes.number,
};

export default WhatsAppIcon;
