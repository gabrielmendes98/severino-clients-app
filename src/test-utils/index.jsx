/* eslint-disable react/prop-types */
import React from 'react';
import { render } from '@testing-library/react-native';
import MainProvider from 'common/contexts/Main';

const Wrapper = ({ children }) => <MainProvider>{children}</MainProvider>;

const customRender = (ui, options) =>
  render(ui, { wrapper: Wrapper, ...options });

export * from '@testing-library/react-native';
export { customRender as render };
