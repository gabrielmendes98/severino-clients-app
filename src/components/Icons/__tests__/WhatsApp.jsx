import React from 'react';
import { rawRender } from 'test-utils';
import WhatsApp from '../WhatsApp';

it('should renders correctly', () => {
  const { toJSON } = rawRender(<WhatsApp />);

  expect(toJSON()).toMatchSnapshot();
});
