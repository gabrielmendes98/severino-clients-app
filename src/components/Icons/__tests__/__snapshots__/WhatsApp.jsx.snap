// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders correctly 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight={18}
  bbWidth={18}
  fill="none"
  focusable={false}
  height={18}
  meetOrSlice={0}
  minX={0}
  minY={0}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      Object {
        "flex": 0,
        "height": 18,
        "width": 18,
      },
    ]
  }
  vbHeight={18}
  vbWidth={18}
  width={18}
  xmlns="http://www.w3.org/2000/svg"
>
  <RNSVGGroup
    fill={null}
    propList={
      Array [
        "fill",
      ]
    }
  >
    <RNSVGPath
      d="M14.25 3.945H3.737v9.538h10.515V3.945Z"
      fill={4294967295}
      propList={
        Array [
          "fill",
        ]
      }
    />
    <RNSVGPath
      clipRule={0}
      d="M14.894 2.497A8.71 8.71 0 0 0 8.754 0C3.968 0 .073 3.82.072 8.515c0 1.501.399 2.966 1.159 4.258L0 17.185l4.602-1.184a8.804 8.804 0 0 0 4.148 1.037h.003c4.784 0 8.679-3.82 8.68-8.516a8.383 8.383 0 0 0-2.539-6.025ZM5.228 8.825c-.108-.142-.885-1.154-.885-2.201 0-1.047.56-1.562.759-1.775a.804.804 0 0 1 .578-.266c.145 0 .29.001.416.007.133.007.312-.05.488.365.18.427.614 1.474.669 1.58.054.107.09.231.018.373s-.109.23-.217.355c-.108.124-.228.278-.325.373-.109.106-.222.221-.096.434.127.213.562.91 1.207 1.474.828.725 1.527.95 1.744 1.056.217.107.343.09.47-.053.127-.142.542-.621.687-.834.145-.213.29-.178.488-.107.199.071 1.265.586 1.482.692.217.107.361.16.416.25.054.088.054.514-.127 1.01-.18.498-1.047.952-1.464 1.012a3.025 3.025 0 0 1-1.366-.084 12.597 12.597 0 0 1-1.237-.448c-2.175-.922-3.596-3.07-3.705-3.213Z"
      fill={4283215696}
      fillRule={0}
      propList={
        Array [
          "fill",
          "fillRule",
        ]
      }
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;
